Grading at 10/29/2023:00:11:36.856316
For commit ID 478fc73e16c6cf6c728b41762ced3b46876e65d8
Grading at Sat Oct 28 08:11:00 PM EDT 2023
Doing some basic checking for your submission

Step 1
Checking that story-step1 compiles
rm -f *~ story-step1 story-step2 story-step3 story-step4 story-step1.o story-step2.o story-step3.o story-step4.o rand_story.o
gcc -std=gnu99 -pedantic -Wall -Werror -ggdb3 -c rand_story.c
gcc -std=gnu99 -pedantic -Wall -Werror -ggdb3 -c story-step1.c
gcc -o story-step1 story-step1.o rand_story.o provided.o
rm story-step1.o
The code compiled (good)
Executable story-step1 exists (good)
Checking with provided test case story.txt
Your file did not match the expected output
Now running YOUR tests for step 1

1.1: Current test is story-step1 story.txt
-----------------------------
It looks like you got different output from us on that case.
Here is the diff (yours on left, ours on right):

cat							      <
cat							      <
cat							      <
Once upon a time, there was a cat who lived in a very cat	Once upon a time, there was a cat who lived in a very cat
cat. 								cat. 
Your file did not match the expected output


1.2: Current test is story-step1
-----------------------------
Great: we both considered this an error
Lets check if it valgrinds:
Valgrind was clean

1.3: Current test is story-step1 story.txt story.txt
-----------------------------
Great: we both considered this an error
Lets check if it valgrinds:
Valgrind was clean

1.4: Current test is story-step1 story2.txt
-----------------------------
It looks like you got different output from us on that case.
Here is the diff (yours on left, ours on right):

cat							      <
cat							      <
cat							      <
cat							      <
cat							      <
cat							      <
cat							      <
Once upon a time, there was a cat. This cat lived in a very	Once upon a time, there was a cat. This cat lived in a very
cat cat. One day, it left its cat cat and met a cat.		cat cat. One day, it left its cat cat and met a cat.
Your file did not match the expected output


1.5: Current test is story-step1 story3.txt
-----------------------------
It looks like you got different output from us on that case.
Here is the diff (yours on left, ours on right):

cat							      <
cat							      <
cat							      <
There is another cat which is in cat then find the correct ca	There is another cat which is in cat then find the correct ca
Your file did not match the expected output


1.6: Current test is story-step1 story4.txt
-----------------------------
Great: we both considered this an error
Lets check if it valgrinds:
Valgrind was clean

1.7: Current test is story-step1 story5.txt
-----------------------------
It looks like you got different output from us on that case.
Here is the diff (yours on left, ours on right):

cat							      <
This is cat should be working					This is cat should be working
Your file did not match the expected output


1.8: Current test is story-step1 story6.txt
-----------------------------
It looks like you got different output from us on that case.
Here is the diff (yours on left, ours on right):

cat							      <
This should not be cat working					This should not be cat working
Your file did not match the expected output


1.9: Current test is story-step1 story7.txt
-----------------------------
It looks like you got different output from us on that case.
Here is the diff (yours on left, ours on right):

cat							      <
There is cat 							There is cat 
Your file did not match the expected output


1.10: Current test is story-step1 story8.txt
-----------------------------
It looks like you got different output from us on that case.
Here is the diff (yours on left, ours on right):

cat							      <
This is cat							This is cat
Your file did not match the expected output


1.11: Current test is story-step1 story9.txt
-----------------------------
It looks like you got different output from us on that case.
Here is the diff (yours on left, ours on right):

cat							      <
This is cat which is invalid.					This is cat which is invalid.
Your file did not match the expected output


1.12: Current test is story-step1 story10.txt
-----------------------------
It looks like you got different output from us on that case.
Here is the diff (yours on left, ours on right):

cat							      <
cat							      <
cat							      <
This is a cat where cat who cat.				This is a cat where cat who cat.
Your file did not match the expected output


1.13: Current test is story-step1 story11.txt
-----------------------------
It looks like you got different output from us on that case.
Here is the diff (yours on left, ours on right):

cat							      <
cat							      <
This is a cat which is cat 					This is a cat which is cat 
Your file did not match the expected output


1.14: Current test is story-step1 story12.txt
-----------------------------
It looks like you got different output from us on that case.
Here is the diff (yours on left, ours on right):

cat							      <
cat							      <
cat							      <
cat							      <
This is a cat that is not a cat supposed to be cat in a cat.	This is a cat that is not a cat supposed to be cat in a cat.
Your file did not match the expected output


1.15: Current test is story-step1 story13.txt
-----------------------------
It looks like you got different output from us on that case.
Here is the diff (yours on left, ours on right):

cat							      <
cat							      <
cat							      <
cat							      <
This is cat in a cat, which is a cat, cat?			This is cat in a cat, which is a cat, cat?
Your file did not match the expected output


1.16: Current test is story-step1 story14.txt
-----------------------------
It looks like you got different output from us on that case.
Here is the diff (yours on left, ours on right):

cat							      <
cat							      <
This is a cat in a cat						This is a cat in a cat
Your file did not match the expected output


1.17: Current test is story-step1 story15.txt
-----------------------------
You considered this a failure case:
timeout: the monitored command dumped core
But we did not.

1.18: Current test is story-step1 story16.txt
-----------------------------
It looks like you got different output from us on that case.
Here is the diff (yours on left, ours on right):

cat							      <
cat							      <
cat							      <
cat							      <
cat							      <
cat							      <
This is a story about a cat, who lives in a cat. This cat is 	This is a story about a cat, who lives in a cat. This cat is 
Your file did not match the expected output


1.19: Current test is story-step1 none.txt
-----------------------------
Your output looks good. Lets check valgrind
Valgrind was clean

1.20: Current test is story-step1 
-----------------------------
Great: we both considered this an error
Lets check if it valgrinds:
Valgrind was clean

Step 2
Checking that story-step2 compiles
rm -f *~ story-step1 story-step2 story-step3 story-step4 story-step1.o story-step2.o story-step3.o story-step4.o rand_story.o
gcc -std=gnu99 -pedantic -Wall -Werror -ggdb3 -c rand_story.c
gcc -std=gnu99 -pedantic -Wall -Werror -ggdb3 -c story-step2.c
gcc -o story-step2 story-step2.o rand_story.o provided.o
rm story-step2.o
The code compiled (good)
Executable story-step2 exists (good)
Now running YOUR tests for step 2

2.1: Current test is story-step2 words.txt
-----------------------------
Your output looks good. Lets check valgrind
Valgrind was clean

2.2: Current test is story-step2 words1.txt
-----------------------------
Your output looks good. Lets check valgrind
Valgrind was clean

2.3: Current test is story-step2 words2.txt
-----------------------------
Your output looks good. Lets check valgrind
Valgrind was clean

2.4: Current test is story-step2 words3.txt
-----------------------------
Your output looks good. Lets check valgrind
Valgrind was clean

2.5: Current test is story-step2 words4.txt
-----------------------------
Your output looks good. Lets check valgrind
Valgrind was clean

2.6: Current test is story-step2 words5.txt
-----------------------------
Your output looks good. Lets check valgrind
Valgrind was clean

2.7: Current test is story-step2 words6.txt
-----------------------------
Your output looks good. Lets check valgrind
Valgrind was clean

2.8: Current test is story-step2 words7.txt
-----------------------------
Great: we both considered this an error
Lets check if it valgrinds:
Valgrind was clean

2.9: Current test is story-step2 words8.txt
-----------------------------
Great: we both considered this an error
Lets check if it valgrinds:
Valgrind was clean

2.10: Current test is story-step2 words9.txt
-----------------------------
Your output looks good. Lets check valgrind
Valgrind was clean

2.11: Current test is story-step2 words10.txt
-----------------------------
Your output looks good. Lets check valgrind
Valgrind was clean

2.12: Current test is story-step2 words.txt words.txt
-----------------------------
Great: we both considered this an error
Lets check if it valgrinds:
Valgrind was clean

Step 3
Checking that story-step3 compiles
rm -f *~ story-step1 story-step2 story-step3 story-step4 story-step1.o story-step2.o story-step3.o story-step4.o rand_story.o
gcc -std=gnu99 -pedantic -Wall -Werror -ggdb3 -c rand_story.c
gcc -std=gnu99 -pedantic -Wall -Werror -ggdb3 -c story-step3.c
gcc -o story-step3 story-step3.o rand_story.o provided.o
rm story-step3.o
The code compiled (good)
Executable story-step3 exists (good)
Now running YOUR tests for step 3

3.1: Current test is story-step3 words.txt story2.txt
-----------------------------
It looks like you got different output from us on that case.
Here is the diff (yours on left, ours on right):

walrus							      <
peculiar						      <
cave							      <
walrus							      <
Once upon a time, there was a walrus. This walrus lived in a 	Once upon a time, there was a walrus. This walrus lived in a 
peculiar cave. One day, it left its peculiar cave and met a w	peculiar cave. One day, it left its peculiar cave and met a w
Your file did not match the expected output


3.2: Current test is story-step3 words.txt story3.txt
-----------------------------
It looks like you got different output from us on that case.
Here is the diff (yours on left, ours on right):

cave							      <
peculiar						      <
There is another cave which is in cave then find the correct 	There is another cave which is in cave then find the correct 
Your file did not match the expected output


3.3: Current test is story-step3 words.txt story4.txt
-----------------------------
Great: we both considered this an error
Lets check if it valgrinds:
Valgrind was clean

3.4: Current test is story-step3 words.txt story5.txt
-----------------------------
Great: we both considered this an error
Lets check if it valgrinds:
Valgrind was clean

3.5: Current test is story-step3 words.txt story6.txt
-----------------------------
Great: we both considered this an error
Lets check if it valgrinds:
Valgrind was clean

3.6: Current test is story-step3 words.txt story7.txt
-----------------------------
Great: we both considered this an error
Lets check if it valgrinds:
Valgrind was clean

3.7: Current test is story-step3 words.txt story8.txt
-----------------------------
Great: we both considered this an error
Lets check if it valgrinds:
Valgrind was clean

3.8: Current test is story-step3 words.txt story9.txt
-----------------------------
Great: we both considered this an error
Lets check if it valgrinds:
Valgrind was clean

3.9: Current test is story-step3 words.txt story10.txt
-----------------------------
Great: we both considered this an error
Lets check if it valgrinds:
Valgrind was clean

3.10: Current test is story-step3 words.txt story11.txt
-----------------------------
Great: we both considered this an error
Lets check if it valgrinds:
Valgrind was clean

3.11: Current test is story-step3 words.txt story12.txt
-----------------------------
It looks like you got different output from us on that case.
Here is the diff (yours on left, ours on right):

cave							      <
cave							      <
peaceful						      <
cave							      <
This is a cave that is not a cave supposed to be peaceful in 	This is a cave that is not a cave supposed to be peaceful in 
Your file did not match the expected output


3.12: Current test is story-step3 words.txt story13.txt
-----------------------------
It looks like you got different output from us on that case.
Here is the diff (yours on left, ours on right):

cave							      <
peculiar						      <
This is cave in a cave, which is a cave, peculiar?		This is cave in a cave, which is a cave, peculiar?
Your file did not match the expected output


3.13: Current test is story-step3 words.txt story14.txt
-----------------------------
Great: we both considered this an error
Lets check if it valgrinds:
Valgrind was clean

3.14: Current test is story-step3 words2.txt story2.txt
-----------------------------
Great: we both considered this an error
Lets check if it valgrinds:
Valgrind was clean

3.15: Current test is story-step3 words4.txt story.txt
-----------------------------
Great: we both considered this an error
Lets check if it valgrinds:
Valgrind was clean

3.16: Current test is story-step3 words3.txt story.txt
-----------------------------
Great: we both considered this an error
Lets check if it valgrinds:
Valgrind was clean

3.17: Current test is story-step3 words6.txt story.txt
-----------------------------
Great: we both considered this an error
Lets check if it valgrinds:
Valgrind was clean

3.18: Current test is story-step3 words7.txt story2.txt
-----------------------------
Great: we both considered this an error
Lets check if it valgrinds:
Valgrind was clean

3.19: Current test is story-step3 words10.txt story2.txt
-----------------------------
Great: we both considered this an error
Lets check if it valgrinds:
Valgrind was clean

3.20: Current test is story-step3 words5 story.txt
-----------------------------
Great: we both considered this an error
Lets check if it valgrinds:
Valgrind was clean

3.21: Current test is story-step3 words none.txt
-----------------------------
Great: we both considered this an error
Lets check if it valgrinds:
Valgrind was clean

Step 4
Checking that story-step4 compiles
rm -f *~ story-step1 story-step2 story-step3 story-step4 story-step1.o story-step2.o story-step3.o story-step4.o rand_story.o
gcc -std=gnu99 -pedantic -Wall -Werror -ggdb3 -c rand_story.c
gcc -std=gnu99 -pedantic -Wall -Werror -ggdb3 -c story-step4.c
gcc -o story-step4 story-step4.o rand_story.o provided.o
rm story-step4.o
The code compiled (good)
Executable story-step4 exists (good)
Now running YOUR tests for step 4

4.1: Current test is story-step4 words.txt story2.txt
-----------------------------
It looks like you got different output from us on that case.
Here is the diff (yours on left, ours on right):

walrus							      <
peculiar						      <
cave							      <
walrus							      <
Once upon a time, there was a walrus. This walrus lived in a 	Once upon a time, there was a walrus. This walrus lived in a 
peculiar cave. One day, it left its peculiar cave and met a w	peculiar cave. One day, it left its peculiar cave and met a w
Your file did not match the expected output


4.2: Current test is story-step4 -n words.txt story2.txt
-----------------------------
It looks like you got different output from us on that case.
Here is the diff (yours on left, ours on right):

walrus							      <
peculiar						      <
cave							      <
dragon							      <
Once upon a time, there was a walrus. This walrus lived in a 	Once upon a time, there was a walrus. This walrus lived in a 
peculiar cave. One day, it left its peculiar cave and met a d	peculiar cave. One day, it left its peculiar cave and met a d
Your file did not match the expected output


4.3: Current test is story-step4 words.txt words.txt
-----------------------------
Your output looks good. Lets check valgrind
Valgrind was clean

4.4: Current test is story-step4 -n words.txt story2.txt
-----------------------------
It looks like you got different output from us on that case.
Here is the diff (yours on left, ours on right):

walrus							      <
peculiar						      <
cave							      <
dragon							      <
Once upon a time, there was a walrus. This walrus lived in a 	Once upon a time, there was a walrus. This walrus lived in a 
peculiar cave. One day, it left its peculiar cave and met a d	peculiar cave. One day, it left its peculiar cave and met a d
Your file did not match the expected output


4.5: Current test is story-step4 -n words.txt story3.txt
-----------------------------
It looks like you got different output from us on that case.
Here is the diff (yours on left, ours on right):

cave							      <
peculiar						      <
There is another cave which is in cave then find the correct 	There is another cave which is in cave then find the correct 
Your file did not match the expected output


4.6: Current test is story-step4 -n words.txt story4.txt
-----------------------------
Great: we both considered this an error
Lets check if it valgrinds:
Valgrind was clean

4.7: Current test is story-step4 -n words.txt story5.txt
-----------------------------
Great: we both considered this an error
Lets check if it valgrinds:
Valgrind was clean

4.8: Current test is story-step4 -n words.txt story6.txt
-----------------------------
Great: we both considered this an error
Lets check if it valgrinds:
Valgrind was clean

4.9: Current test is story-step4 -n words.txt story7.txt
-----------------------------
Great: we both considered this an error
Lets check if it valgrinds:
Valgrind was clean

4.10: Current test is story-step4 -n words.txt story8.txt
-----------------------------
Great: we both considered this an error
Lets check if it valgrinds:
Valgrind was clean

4.11: Current test is story-step4 -n words.txt story9.txt
-----------------------------
Great: we both considered this an error
Lets check if it valgrinds:
Valgrind was clean

4.12: Current test is story-step4 -n words.txt story10.txt
-----------------------------
Great: we both considered this an error
Lets check if it valgrinds:
Valgrind was clean

4.13: Current test is story-step4 -n words.txt story11.txt
-----------------------------
Great: we both considered this an error
Lets check if it valgrinds:
Valgrind was clean

4.14: Current test is story-step4 -n words.txt story12.txt
-----------------------------
Great: we both considered this an error
Lets check if it valgrinds:
Valgrind was clean

4.15: Current test is story-step4 -n words.txt story13.txt
-----------------------------
It looks like you got different output from us on that case.
Here is the diff (yours on left, ours on right):

cave							      <
peculiar						      <
This is cave in a cave, which is a cave, peculiar?		This is cave in a cave, which is a cave, peculiar?
Your file did not match the expected output


4.16: Current test is story-step4 -n words.txt story14.txt
-----------------------------
Great: we both considered this an error
Lets check if it valgrinds:
Valgrind was clean

4.17: Current test is story-step4 -n words2.txt story2.txt
-----------------------------
Great: we both considered this an error
Lets check if it valgrinds:
Valgrind was clean

4.18: Current test is story-step4 -n words4.txt story.txt
-----------------------------
Great: we both considered this an error
Lets check if it valgrinds:
Valgrind was clean

4.19: Current test is story-step4 -n words3.txt story.txt
-----------------------------
Great: we both considered this an error
Lets check if it valgrinds:
Valgrind was clean

4.20: Current test is story-step4 -n words6.txt story.txt
-----------------------------
Great: we both considered this an error
Lets check if it valgrinds:
Valgrind was clean

4.21: Current test is story-step4 -n words7.txt story2.txt
-----------------------------
Great: we both considered this an error
Lets check if it valgrinds:
Valgrind was clean

4.22: Current test is story-step4 -n words5 story.txt
-----------------------------
Great: we both considered this an error
Lets check if it valgrinds:
Valgrind was clean

4.23: Current test is story-step4 -n words none.txt
-----------------------------
Great: we both considered this an error
Lets check if it valgrinds:
Valgrind was clean

4.24: Current test is story-step4 -n words5.txt story2.txt
-----------------------------
Great: we both considered this an error
Lets check if it valgrinds:
Valgrind was clean

4.25: Current test is story-step4 -n
-----------------------------
Great: we both considered this an error
Lets check if it valgrinds:
Valgrind was clean

4.26: Current test is story-step4 -n words.txt story15.txt
-----------------------------
Great: we both considered this an error
Lets check if it valgrinds:
Valgrind reported errors:
==1523070== Memcheck, a memory error detector
==1523070== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==1523070== Using Valgrind-3.18.1 and LibVEX; rerun with -h for copyright info
==1523070== Command: ./story-step4 -n words.txt story15.txt
==1523070== Parent PID: 1523069
==1523070== 
==1523070== Invalid read of size 1
==1523070==    at 0x48AB580: ____strtol_l_internal (strtol_l.c:291)
==1523070==    by 0x1095D0: isPositiveInt (rand_story.c:10)
==1523070==    by 0x109B70: parsing (rand_story.c:121)
==1523070==    by 0x109526: main (story-step4.c:38)
==1523070==  Address 0x0 is not stack'd, malloc'd or (recently) free'd
==1523070== 
==1523070== 
==1523070== Process terminating with default action of signal 11 (SIGSEGV)
==1523070==  Access not within mapped region at address 0x0
==1523070==    at 0x48AB580: ____strtol_l_internal (strtol_l.c:291)
==1523070==    by 0x1095D0: isPositiveInt (rand_story.c:10)
==1523070==    by 0x109B70: parsing (rand_story.c:121)
==1523070==    by 0x109526: main (story-step4.c:38)
==1523070==  If you believe this happened as a result of a stack
==1523070==  overflow in your program's main thread (unlikely but
==1523070==  possible), you can try to increase the size of the
==1523070==  main thread stack using the --main-stacksize= flag.
==1523070==  The main thread stack size used in this run was 8388608.
==1523070== 
==1523070== HEAP SUMMARY:
==1523070==     in use at exit: 4,793 bytes in 17 blocks
==1523070==   total heap usage: 131 allocs, 114 frees, 10,196 bytes allocated
==1523070== 
==1523070== For a detailed leak analysis, rerun with: --leak-check=full
==1523070== 
==1523070== For lists of detected and suppressed errors, rerun with: -s
==1523070== ERROR SUMMARY: 1 errors from 1 contexts (suppressed: 0 from 0)

4.27: Current test is story-step4 -n words.txt story16.txt
-----------------------------
It looks like you got different output from us on that case.
Here is the diff (yours on left, ours on right):

walrus							      <
cave							      <
peaceful						      <
dragon							      <
This is a story about a walrus, who lives in a cave. This wal	This is a story about a walrus, who lives in a cave. This wal
Your file did not match the expected output


You have NOT passed your test cases!
You should go and figure out/understand where thing went wrong
And fix the problem(s) before the deadline

Overall Grade: NO GRADE YET
